internal/services/apimanagement/api_management_policy_fragment_resource.go

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apimanagement
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/apimanagement/2022-08-01/policyfragment"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+func resourceApiManagementPolicyFragment() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceApiManagementPolicyFragmentCreate,
+		Read:   resourceApiManagementPolicyFragmentRead,
+		Update: resourceApiManagementPolicyFragmentUpdate,
+		Delete: resourceApiManagementPolicyFragmentDelete,
+		Importer: pluginsdk.ImporterValidatingResourceIdThen(func(id string) error {
+			_, err := policyfragment.ParsePolicyFragmentIDInsensitively(id)
+			return err
+		}, func(ctx context.Context, d *pluginsdk.ResourceData, meta interface{}) ([]*pluginsdk.ResourceData, error) {
+			client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+
+			id, err := policyfragment.ParsePolicyFragmentIDInsensitively(d.Id())
+			if err != nil {
+				return nil, err
+			}
+
+			resp, err := client.Get(ctx, *id, policyfragment.GetOperationOptions{
+				Format: pointer.To(policyfragment.PolicyFragmentContentFormatXml),
+			})
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return nil, fmt.Errorf("Api Management Policy Fragment %q was not found in Api Management instance %q in Resource Group %q", id.PolicyFragmentName, id.ServiceName, id.ResourceGroupName)
+				}
+
+				return nil, fmt.Errorf("retrieving Api Management Policy Fragment %q (Api Management: %q, Resource Group %q): %+v", id.PolicyFragmentName, id.ServiceName, id.ResourceGroupName, err)
+			}
+
+			d.Set("format", policyfragment.PolicyFragmentContentFormatXml)
+
+			return []*pluginsdk.ResourceData{d}, nil
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": schemaz.SchemaApiManagementChildName(),
+
+			"resource_group_name": commonschema.ResourceGroupName(),
+
+			"api_management_name": schemaz.SchemaApiManagementName(),
```

Comment:
Since we're dependent on the API Management resource here, it'd be better to make this:

```suggestion
			"api_management_id": commonschema.ResourceIDReferenceRequiredForceNew(&apimanagementservice.ServiceId{}),
```

since that means this resource will get recreated should the parent APIM get recreated - as such can we update this?